### PR TITLE
feat(MarketsOffers): Implement the market offers table.

### PR DESCRIPTION
**Pull request by abouolia**

### CODE REVIEW

File: packages/renderer/src/components/organisms/MarketsTransactions/MarketsTransactions.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// =============================================================================
+
+import { Loader } from "@mantine/core";
+import { useMarketsOffers } from "@hooks/haveno/useMarketsOffers";
+import { MarketTransactionsTable } from "@molecules/MarketTransactionsTable";
+import type { FC } from "react";
+
+export function MarketsTransactionsLoaded() {
```
Comment (abouolia) [inline packages/renderer/src/components/organisms/MarketsTransactions/MarketsTransactions.tsx:22]: Replace to `MarketsOffersLoaded`.

File: packages/renderer/src/components/atoms/Currency/Currency.tsx [hunk 2]
```diff
@@ -15,15 +15,19 @@
...
 import { useMemo } from "react";
+import type { FormatNumberOptions } from "react-intl";
 import { useIntl } from "react-intl";
 
-interface CurrencyProps {
+type CurrencyFormatType = "symbol" | "code" | "name" | "narrowSymbol";
+
+interface CurrencyProps extends FormatNumberOptions {
   currencyCode?: string;
+  currentDisplay?: CurrencyFormatType;
```
Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Currency/Currency.tsx]: `currentDisplay` isn't a good choice for a name. How about `format`?

Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Currency/Currency.tsx]: `currentDisplay` isn't a good choice for a name. How about `format`?

File: packages/renderer/src/components/atoms/Percentage/Percentage.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// =============================================================================
+
+interface PercentageProps {
+  value: number;
+}
+
+export function Percentage({ value }: PercentageProps) {
+  return <>%{value * 100}</>;
```
Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Percentage/Percentage.tsx]: Is this the correct format? Percentage values are supposed to look like `3.45%`, `25%`, etc

Comment (schowdhuri) [inline packages/renderer/src/components/atoms/Percentage/Percentage.tsx]: 1. Is this the correct format? Percentage values are supposed to look like `3.45%`, `25%`, etc.
2. this doesn't need to be a component. A util method like `fractionToPercent` is far simpler and more descriptive

File: packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  MarketOffersPaymentCell,
+} from "./MarketOffersTableCell";
+import { Table } from "@molecules/Table";
+import { LangKeys } from "@constants/lang";
+import { TableVariant } from "@molecules/Table/_types";
+
+const table = createTable().setRowType<MarketOffer>();
+
+interface MarketOffersTableProps {
+  data: MarketOffer[];
```
Comment (schowdhuri) [inline packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx]: Global comment: `Array<Foo>` notation. This has been a recurring issue. Please make a note of this for all future work.

Comment (schowdhuri) [inline packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx]: Global comment: `Array<Foo>` notation. This has been a recurring issue. Please make a note for future work.

File: packages/renderer/src/components/organisms/MarketsOffers/_utils.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// =============================================================================
+
+import { MarketOfferPaymentMethod } from "@molecules/MarketOffersTable";
+import type { MarketOfferData } from "@hooks/haveno/useMarketsOffers";
+import type { MarketOffer } from "@molecules/MarketOffersTable";
+
+export const transformToMarketsOffers = (
+  offers: MarketOfferData[]
```
Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts]: `Array<Foo>` notation

Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts]: `Array<Foo>` notation

File: packages/renderer/src/components/organisms/MarketsOffers/_utils.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+): MarketOffer[] => {
```
Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts]: `Array<Foo>` notation

Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts]: `Array<Foo>` notation

File: packages/renderer/src/components/organisms/MarketsOffers/_utils.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  return offers.map((offer) => ({
+    price: offer.price,
+    priceCurrency: offer.counterCurrencyCode,
+    amount: offer.amount,
+    amountCurrency: offer.baseCurrencyCode,
+    costCurrency: offer.baseCurrencyCode,
+    priceComparison: 0.1,
```
Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:31]: Are these constants: `priceComparison`, `cost`?

Comment (schowdhuri) [inline packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:31]: Are these constants: `priceComparison`, `cost`?

File: packages/renderer/src/components/templates/NavbarLayout/index.tsx [hunk 8]
```diff
@@ -14,15 +14,27 @@
...
-import type { FC } from "react";
 import { Sidebar } from "@organisms/Sidebar";
 
-export const NavbarLayout: FC = (props) => {
-  const { children } = props;
-  const { classes } = useStyles();
+interface NavbarLayoutProps extends DefaultProps {
+  children: ReactNode;
+}
+export const NavbarLayout = ({
```
Comment (schowdhuri) [inline packages/renderer/src/components/templates/NavbarLayout/index.tsx]: Use `FC` and remove `  children: ReactNode`

Comment (schowdhuri) [inline packages/renderer/src/components/templates/NavbarLayout/index.tsx]: Newline required

File: packages/renderer/src/hooks/haveno/useMarketsOffers.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+interface MarketsOfferesQuery {
+  assetCode: string;
+  direction?: "buy" | "sell";
+}
+
+export function useMarketsOffers(query: MarketsOfferesQuery) {
+  const client = useHavenoClient();
+
+  return useQuery<MarketOfferData[], Error>(
```
Comment (schowdhuri) [inline packages/renderer/src/hooks/haveno/useMarketsOffers.ts]: `Array<Foo>` notation

Comment (schowdhuri) [inline packages/renderer/src/hooks/haveno/useMarketsOffers.ts]: `Array<Foo>` notation

File: packages/renderer/src/hooks/haveno/useMarketsOffers.ts [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    [QueryKeys.MarketsOffers, query],
+    async () => {
+      const offers = await client.getMyOffers(query.assetCode);
+      return transformData(offers);
+    }
+  );
+}
+
+const transformData = (offers: OfferInfo[]) => {
```
Comment (schowdhuri) [inline packages/renderer/src/hooks/haveno/useMarketsOffers.ts]: `Array<Foo>` notation

Comment (schowdhuri) [inline packages/renderer/src/hooks/haveno/useMarketsOffers.ts]: `Array<Foo>` notation

**Review by schowdhuri — CHANGES_REQUESTED** [review]
- remove Account Age and Number of trades from the table
- clicking on the `Market` link doesn't do anything
- let's upgrade `MarketOffersTable` to an organism
- `Array<Foo>` notation must be used everywhere

File: packages/renderer/src/components/templates/NavbarLayout/index.tsx (hunk 8 above)
Comment (abouolia) [reply packages/renderer/src/components/templates/NavbarLayout/index.tsx]: `NavbarLayoutProps ` extends to `DefaultProps`, so `NavbarLayout ` has more than `children` prop.

File: packages/renderer/src/components/organisms/MarketsOffers/_utils.ts (hunk 7 above)
Comment (abouolia) [reply packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:31]: Ok mapped the cost to `txFees` but we haven't price comparison or any percentage value.

File: packages/renderer/src/components/atoms/Percentage/Percentage.tsx (hunk 3 above)
Comment (abouolia) [reply packages/renderer/src/components/atoms/Percentage/Percentage.tsx]: Agree with you, Removed percentage as a component and relayed on a small utility. btw all percentages data should be stored as fractions.

**Comment by abouolia** [comment]
- Hided the Account Age and Number of trades columns from the table.
- fixed: clicking on the Market link doesn't do anything.
- fixed: let's upgrade MarketOffersTable to an organism.
- fixed: Array<Foo> notation must be used everywhere

File: packages/renderer/src/components/atoms/Currency/Currency.tsx (hunk 2 above)
Comment (abouolia) [reply packages/renderer/src/components/atoms/Currency/Currency.tsx]: I don't know how did that. 🤦 🤦

File: packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx (hunk 4 above)
Comment (abouolia) [reply packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx]: Remembered all the recurring mistakes except this one. :D

File: packages/renderer/src/components/templates/NavbarLayout/index.tsx (hunk 8 above)
Comment (schowdhuri) [reply packages/renderer/src/components/templates/NavbarLayout/index.tsx]:
```tsx
const NavbarLayout: FC<DefaultProps> = (props) => {
 // ...
}
```

**Review by schowdhuri — CHANGES_REQUESTED** [review]
Thanks! This is nearly done. Just one tiny change and we'll be good to go!

File: packages/renderer/src/components/atoms/Currency/Currency.tsx (hunk 2 above)
Comment (abouolia) [reply packages/renderer/src/components/atoms/Currency/Currency.tsx]: This is outdated, i fixed that already.

**Review by schowdhuri — APPROVED** [review]
Looks perfect! 🚀

**Pull request merged**